src/sail_json_backend/json.ml

```diff
@@ -231,20 +232,69 @@ let parse_encdec i mc format =
       end
   | _ -> assert false
 
+let filter_non_operands components =
+  let rec aux acc = function
+    | [] -> List.rev acc
+    | hd :: tl -> if String.trim hd = "spc" then tl else aux (hd :: acc) tl
+  in
+  aux [] components
+
+let extract_operands k regex components =
+  let rec aux acc = function
+    | [] -> List.rev acc
+    | hd :: tl ->
+        if Str.string_match regex hd 0 then (
+          let operand = Str.matched_group 1 hd in
+          try
+            let comma_index = String.index operand ',' in
+            debug_print ("Operand before trimming: " ^ operand);
+            let trimmed = String.sub operand 0 comma_index in
+            debug_print ("Final trimmed operand: " ^ trimmed);
+            let inputl = Hashtbl.find inputs k in
+            if List.mem trimmed inputl then aux (trimmed :: acc) tl else aux acc tl
```

Comment:
Can you make this a more generic loop that looks at each element in the list of parameters of this function? For example, if it encounters a string "func(a,b,c)", it checks to see if "a", "b", and "c" are in "inputs"?